lang/c++/api/Node.hh

```diff
@@ -152,6 +152,11 @@ class AVRO_DECL Node : private boost::noncopyable
 
     virtual void setLeafToSymbolic(int index, const NodePtr &node) = 0;
 
+    // Serialize the default value GenericDatum g for the node contained
```

Comment:
I suggest using a Visitor pattern instead of adding this functionality directly into the Node.